### PR TITLE
[JENKINS-51248] Added label and container regex to fail job if bad

### PR DESCRIPTION
**Pull request by MattLud**
Initial approach to providing validation and immediate job failure on bad label and name. 

Please review correctness of regex and if throwable Exception should be changed.

### CODE REVIEW

**Comment by carlossg** [comment]
I don't like the fact that a lot of setters throw exceptions and get propagated all over the place

I have asked around and got this advice

> throw `IllegalArgumentException` from `@DataBoundConstructor` / `@DataBoundSetter`
> matched with `StepDescriptor.doCheckXXX` for the *Pipeline Syntax* UI
> you can also just throw (say) `AbortException` from `Step.start`

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java [hunk 1]
```diff
@@ -81,6 +81,10 @@ public ContainerTemplate(String name, String image, String command, String args)
 
     @DataBoundSetter
     public void setName(String name) {
+        if(!PodTemplateUtils.validateContainerName(name))
+        {
+            throw new IllegalArgumentException(String.format("Container Names MUST match RFC 1123 - They can only contain lowercase letters, numbers or dashes. ContainerName-%s",name));
```
[the remark this answers is not in the document]
Comment (MattLud) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java]: Added. Thanks!

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 2]
```diff
@@ -355,7 +355,11 @@ public String getActiveDeadlineSecondsStr() {
     }
 
     @DataBoundSetter
-    public void setLabel(String label) {
+    public void setLabel(String label) throws Exception {
+        if(!PodTemplateUtils.validateLabel(label))
+        {
+            throw new Exception(String.format("Labels must follow required specs - https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set - %s", label));
```
[the remark this answers is not in the document]
Comment (MattLud) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Correct  - removed it and replaced with above advice.

**Comment by MattLud** [comment]
Updated to remove grossy unnecessary exception declarations.

**Pull request merged**